examples/wasm/iwasmt.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * Copyright (C) 2020 TU Bergakademie Freiberg Karl Fessel
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+//
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include <string.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#include "wasm_export.h"
+#pragma GCC diagnostic pop
+
+#include <thread.h>
+
+/*provide some test program*/
+#include "blob/main.wasm.h"
+
+
+#define DEFAULT_THREAD_STACKSIZE (6 * 1024)
+#define DEFAULT_THREAD_PRIORITY 50
+
+static int app_argc;
+static char **app_argv;
+
+static void*
+app_instance_main(wasm_module_inst_t module_inst)
+{
+    const char *exception;
+
+    wasm_application_execute_main(module_inst, app_argc, app_argv);
+    if ((exception = wasm_runtime_get_exception(module_inst))){
+        puts(exception);
+    }
+    return NULL;
+}
+
+
+void *
+iwasm_t(void *arg1)
+{
+    wasm_module_t wasm_module = (wasm_module_t) arg1;
+    wasm_module_inst_t wasm_module_inst = NULL;
+    char error_buf[128];
+
+    /* instantiate the module */
+    if (!(wasm_module_inst = wasm_runtime_instantiate(wasm_module, 8 * 1024,
+        8 * 1024, error_buf, sizeof(error_buf)))) {
+        puts(error_buf);
+    }else{
+        app_instance_main(wasm_module_inst);
+        /* destroy the module instance */
+        wasm_runtime_deinstantiate(wasm_module_inst);
+    }
+    return NULL;
+}
+
+void *
+iwasm_main(void *arg1)
+{
+    (void) arg1; /*unused*/
+    uint8_t *wasm_file_buf = NULL;
+    unsigned wasm_file_buf_size = 0;
+    wasm_module_t wasm_module = NULL;
+    char error_buf[128];
+
+    RuntimeInitArgs init_args;
+
+//chose allocator non defaults to system allocator
+#define FUNC_ALLOC
+//#define POOL_ALLOC
+
+    memset(&init_args, 0, sizeof(RuntimeInitArgs));
+#ifdef POOL_ALLOC
+    static char global_heap_buf[256 * 1024] = { 0 };//(256 kB)
+
+    init_args.mem_alloc_type = Alloc_With_Pool;
+    init_args.mem_alloc_option.pool.heap_buf = global_heap_buf;
+    init_args.mem_alloc_option.pool.heap_size = sizeof(global_heap_buf);
+#elif defined(FUNC_ALLOC)
+    init_args.mem_alloc_type = Alloc_With_Allocator;
+/*deactivate Wpedantic for some lines*/
+#pragma GCC diagnostic ignored "-Wpedantic"
+    init_args.mem_alloc_option.allocator.malloc_func = malloc;
+    init_args.mem_alloc_option.allocator.realloc_func = realloc;
+    init_args.mem_alloc_option.allocator.free_func = free;
+#pragma GCC diagnostic pop
+#else
+    init_args.mem_alloc_type = Alloc_With_System_Allocator;
+#endif
```

Comment:
IMO you could just use the variant using `malloc()` and friends here. In the end, the RIOT examples for packages are not tutorial in how to use the external packages, but in how to use the package with RIOT. Setting up an alternative allocation scheme is something that is (hopefully) covered by the upstream documentation of the package and requires no RIOT specific special handling.

For that reason, I think it is better to keep the example as simple and minimal as possible, except when showing important RIOT specifics.